app/dojo.py

```diff
@@ -349,5 +378,146 @@ def load_people(self, filename):
 			+ "_" * 60 + "\n\n" + output + "\n" +text_format.CEND)
 
+	def save_state(self, db_name):
+		"""Saves all data from the app to a specified 
+		database
 
-		
\ No newline at end of file
+		"""
+
+		engine = create_engine('sqlite:///' + db_name + '.db')
+		Base.metadata.bind = engine
+		Base.metadata.create_all(engine)
+		DBSession = sessionmaker(bind=engine)
+		session = DBSession()
+
+		for room in itertools.chain(self.all_offices,self.all_livingspace):
+			save_room = RoomModel(
+				name=room.name,
+				room_type=room.room_type,
+				capacity=room.capacity,
+				)
+			existing = session.query(RoomModel).filter(
+				RoomModel.name==room.name).count()
+			if not existing:
+				session.add(save_room)
+			session.commit()
+
+		for person in itertools.chain(self.all_staff,self.all_fellows):
+			livingspace_waiting = [each_person for each_person\
+			 	 in self.livingspace_waitinglist if each_person == person]
+			if person.role == "FELLOW" and person.wants_accomodation == "y":
+				for living in self.all_livingspace:
+					living_allocated = [occupant for occupant in living.occupants\
+					 if occupant == person]
+					if living_allocated:
+						living_status = living.name
+				if livingspace_waiting:
+					living_status = "UNALLOCATED"		
+			else:
+				living_status = "N/A"			 	 
+
+			if self.all_offices:	
+				for office in self.all_offices:
+					office_allocated = [occupant for occupant\
+					 in office.occupants if occupant == person]
+					if office_allocated:
+						office_status = office.name
+			else:
+				office_status = "UNALLOCATED"	
+
+			save_person = PersonModel(
+				name=person.name,
+				email=person.email,
+				role=person.role,
+				office_space=office_status,
+				living_space=living_status
+				)
+			existing = session.query(PersonModel).filter(
+				PersonModel.email==person.email).count()
+			if not existing:
+				session.add(save_person)
+		session.commit()
+
+		output = ("\n\nApplication data successfully saved to database!\n\n")
+		return (text_format.CBOLD + output + text_format.CEND)
+
+	
+	def load_state(self, db_name):
+		"""This method loads data from the db
+		into the application
+		"""
+		if not os.path.isfile("{}.db".format(db_name)):
+			return (text_format.CRED + "\nThe database {}.db does not exist!\n"
+				.format(db_name) 
+				+text_format.CEND)
+			
+		engine = create_engine('sqlite:///' + db_name + '.db')
+		Base.metadata.bind = engine			
+		DBSession = sessionmaker(bind=engine)
+		session = DBSession()
+			
+		room_list = session.query(RoomModel).all()
+		person_list = session.query(PersonModel).all()	
+		#load persons
+		for person in person_list:
+			if person.role == "STAFF":
+				load_staff = Staff(person.name, person.email)
+				self.all_staff.append(load_staff)
+			if person.role == "FELLOW" and person.living_space == "N/A":
+				load_fellow = Fellow(person.name, person.email, "N")
+				self.all_fellows.append(load_fellow)
+			if person.role == "FELLOW" and person.living_space != "N/A":
```

Comment:
I had considered doing that, however, it would conflict with the db data since the column living space in the db holds the living space name or "N/A" to indicate that accomodation is not applicable